src/server/editorServices.ts

```diff
@@ -754,6 +756,106 @@ namespace ts.server {
             return false;
         }
 
+        acquireTypingForJs(project: Project) {
+            if (!project || !this.host.getTsd) {
+                return;
+            }
+            const tsd = this.host.getTsd();
+            if (tsd === undefined) {
+                return;
+            }
+
+            const typingOptions: TypingOptions = project.projectOptions
+                ? project.projectOptions.typingOptions
+                : { enableAutoDiscovery: true };
+            const compilerOptions: CompilerOptions = project.projectOptions
+                ? project.projectOptions.compilerOptions
+                : undefined;
+            const projectRootPath: Path = project.isConfiguredProject() ? getDirectoryPath(<Path>project.projectFilename) : undefined;
+
+            // Todo: add support for local cache path
+            const cachePath = this.host.globalTypingCachePath;
+            const tsdJsonPath = toPath("tsd.json", cachePath, getCanonicalFileName);
+            const tsdJson = this.host.getTsdJson(tsdJsonPath);
+            const { cachedTypingPaths, newTypingNames } = JsTyping.discoverTypings(
+                sys,
+                project.getFileNames(),
+                projectRootPath,
+                /*safeListPath*/ undefined,

```

Comment:
Yes, as we haven't figured out a proper way to host / download the safe list yet.